### PR TITLE
Fixed index type integration tests [ECR-3415]

### PR DESCRIPTION
**Pull request by skletsun**
## Overview
Fixed integration tests that check the type of index.
---
See: https://jira.bf.local/browse/ECR-3415

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java [hunk 1]
```diff
@@ -197,7 +196,8 @@ void indexConstructorPersistsIndexTypeInfo() throws CloseFailuresException {
       Exception e = assertThrows(RuntimeException.class, () -> createOfOtherType(name, snapshot));
 
       // TODO: Change message after https://jira.bf.local/browse/ECR-3354
-      assertThat(e, hasMessage(containsString("Index type doesn't match specified")));
+      assertThat(e, hasMessage(containsStringIgnoringCase(
+              "Index type does not match specified one")));
```
Comment (skletsun) [inline exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: After ECR-3418 we have some changes in error messages. For the Entry case it is now _"Error while reading the index state. Possibly the index type does not match specified one"_, for the rest of indices it is _"Index type does not match specified one"_. @dmitry-timofeev is this solution acceptable?

Comment (vitvakatu) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: I think it is not the solution of the initial problem, as it does not show even the actual type of the index, does it?

Comment (skletsun) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: Well, at line 198 there is a //TODO with link to the task I created already for this problem - https://jira.bf.local/browse/ECR-3354

Comment (vitvakatu) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: Ah, sorry, I thought that this PR is actually aimed to fix ECR-3354 :D

Comment (skletsun) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: What we can see as an error output for tests on indices other than `Entry`:
```
"assertion failed: `(left == right)`
  left: `KeySet`,
 right: `List`: Index type does not match specified one"
```
So, in fact one can see the index type, but as part of a wrong assertion.

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: I think yes, the alternative is too complicated to bother (add an assumption here that index type is not an entry to exclude it + an entry-only test).

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: Do those left and right get into the message?

Comment (skletsun) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: Yes, it is the message we get from assertion in the `metadata::index_metadata()`:
```
        assert_eq!(
            metadata.index_type, index_type,
            "Index type does not match specified one"
        );
```

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: and into the message of a Java Exception?

Comment (skletsun) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: Yes. `Index::new()` obviously panics on native assert -> we catch it and throw as a `RuntimeException`.

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: So ECR-3354 is to make it clear which type is expected?

Comment (skletsun) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java]: I don't remember the details of the discussion that caused this ticket, but in general - yes.

**Pull request merged**